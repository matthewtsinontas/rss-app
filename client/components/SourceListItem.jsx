import React from 'react';

/**
  Component to render a list item for a source that you have subscribed to
  Simple list item which displays a source and attaches a "delete" button which
    can fire an action
*/
const SourceListItem = ({source, deleteItem}) => (
  <li>
    {source}
    <button className="btn btn-danger" onClick={deleteItem}>
      <span className="glyphicon glyphicon-trash" aria-hidden="true"></span>
    </button>
  </li>
);

export default SourceListItem;
