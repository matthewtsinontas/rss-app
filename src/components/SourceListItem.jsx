import React from 'react';

const SourceListItem = ({source, deleteItem}) => (
  <li>{source} - <a onClick={deleteItem}>delete</a></li>
);

export default SourceListItem;
