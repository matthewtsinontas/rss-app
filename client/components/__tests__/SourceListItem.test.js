import React from 'react';
import ReactDOM from 'react-dom';
import SourceListItem from '../SourceListItem.jsx';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

const item = <SourceListItem source="test" deleteItem={() => {}}/>;

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(item, div);
});

it('renders with the correct props', () => {
  const tree = renderer.create(item).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call the provided function when delete is clicked', () => {
  const onDeleteClick = sinon.spy();
  const wrapper = shallow(
    <SourceListItem source="text" deleteItem={onDeleteClick} />
  );
  wrapper.find('button').simulate('click');
  expect(onDeleteClick.calledOnce).toEqual(true);
});
