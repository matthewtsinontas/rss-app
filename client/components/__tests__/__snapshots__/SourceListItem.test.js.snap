// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct props 1`] = `
<li>
  test
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-trash"
    />
  </button>
</li>
`;
